### ISSUE TITLE
performance regression

### ISSUE THREAD

**Issue by SirZach** [switchfly/ember-cli-mocha#62]
I updated ember-cli-mocha from 0.4.2 to 0.8.0 and noticed a huge performance regression.

```
Build successful - 63094ms.

Slowest Trees                                 | Total
----------------------------------------------+---------------------
JSHint tests- Mocha                           | 40829ms
Babel                                         | 5971ms
JSHint app- Mocha                             | 4122ms

Slowest Trees (cumulative)                    | Total (avg)
----------------------------------------------+---------------------
JSHint tests- Mocha (1)                       | 40829ms
Babel (5)                                     | 9657ms (1931 ms)
JSHint app- Mocha (1)                         | 4122ms
```

Any thoughts?

**Comment by SirZach** [comment]
bump

**Comment by SirZach** [comment]
bump

**Comment by rwjblue** [comment]
This isn't really very actionable.  There are many many factors involved in your application, and none of us have the insight into what is going on that you do.

I'd suggest going through the changes, and testing each one (essentially like a `git bisect`) to figure out what introduced these issues in your application:

https://github.com/switchfly/ember-cli-mocha/compare/v0.4.2...v0.8.0

Once we know what introduced it, we might have more information to help you.

**Comment by SirZach** [comment]
When using <code>"ember-cli-mocha": "switchfly/ember-cli-mocha#2a7bcc8"</code> that's the first commit that starts the regression. With <code>"ember-cli-mocha": "switchfly/ember-cli-mocha#e5427d6"</code>, (the commit right before the above mentioned commit), my compile time is
```shell
Build successful - 27184ms.

Slowest Trees                                 | Total
----------------------------------------------+---------------------
Babel                                         | 7088ms
JSHint app- Mocha                             | 4094ms
Concat: Vendor                                | 3134ms
Babel                                         | 3054ms
ES6: App Tree                                 | 2170ms
JSHint tests- Mocha                           | 1456ms
LessCompiler                                  | 1397ms

Slowest Trees (cumulative)                    | Total (avg)
----------------------------------------------+---------------------
Babel (6)                                     | 11394ms (1899 ms)
JSHint app- Mocha (1)                         | 4094ms
Concat: Vendor (1)                            | 3134ms
ES6: App Tree (1)                             | 2170ms
LessCompiler (2)                              | 1537ms (768 ms)
JSHint tests- Mocha (1)                       | 1456ms
```

Effectively making https://github.com/switchfly/ember-cli-mocha/commit/e5427d66d7ed987d9178c9e7df00b4074183a53c the commit that started the slowdown.

**Comment by rwjblue** [comment]
That was added in https://github.com/switchfly/ember-cli-mocha/pull/55, which allows the broccoli-jshint dependency to float.

You can lock your project's broccoli-jshint version down, and start narrowing down which of the upstream commits introduced the regression for you.  

https://github.com/rwjblue/broccoli-jshint/compare/v0.5.3...v0.5.6

**Comment by SirZach** [comment]
kind of a silly question but how do i lock down the dependency version for a node_module that's brought in from another node_module?

**Comment by SirZach** [comment]
https://github.com/rwjblue/broccoli-jshint/commit/12d865766168175800e702b1bec2d123548da7a2 is the commit that caused the slow-down

**Comment by rwjblue** [comment]
Not silly at all. Installing a dependency at the project will override nested child dependencies.

So if you run:

```
npm install --save-dev broccoli-jshint@0.5.3
rm -rf node_modules && npm cache clear && npm install
```

`broccoli-jshint` will be used by any deeply nested child dependencies directly from the project's `node_modules/` dir.  Then you can tweak that (or `npm link` it if you want to bisect).

**Comment by rwjblue** [comment]
Ahh, ok. So its actually a performance regression in JSHint itself.  What version of JSHint are you getting inside `broccoli-jshint`'s `node_modules`? 

You can see if things are fixed with `^2.6.2` (which would also include 2.7.0 and 2.8.0). 

https://github.com/jshint/jshint/compare/2.6.2...2.8.0

**Comment by SirZach** [comment]
<code>2.8.0</code> still had the same slowness. Somewhere between <code>2.5.0</code> and <code>2.6.2</code> is where the regression lies but there are over 250 commits between those two versions. I'll create/search for issues there. Thanks for your help!
